src/client/legacy/connect/dns.rs

```diff
@@ -118,7 +118,9 @@ impl Service<Name> for GaiResolver {
     }
 
     fn call(&mut self, name: Name) -> Self::Future {
+        let current_span = tracing::Span::current();
         let blocking = tokio::task::spawn_blocking(move || {
+            let _enter = current_span.enter();
```

Comment:
The only question I have is since the call below is blocking, does entering the span the whole time mess with measurements?

cc @hawkw

Reply:
That's a good thing though right? Is there a reason that the the timing of a http request shouldn't include DNS lookup time. Are you concerned with existing measurement metrics looking bad because they now include this time?

I think it might be worth creating a child debug span here for the DNS resolution (done in 0f690ae). This allows metrics for just dns resolution to be visible when the debug span level is enabled:

debug enabled (dns time is not included in the main span):
```shell
RUST_LOG=debug cargo run --example=client --features=full http://example.com
```
<img width="1638" alt="image" src="https://github.com/user-attachments/assets/908e4e84-8b9f-41c5-9fdb-f7b655f7fb82">


debug disabled (dns time is included in the main span):
```shell
RUST_LOG=info cargo run --example=client --features=full http://example.com
```
<img width="1052" alt="image" src="https://github.com/user-attachments/assets/7a678610-75fe-48c2-9ca6-56d7fc198256">

Reply:
Ah ok, cool. I didn't explain what I meant well, I was worried that since the span would be in a "blocking" task, I worried that the overall span could appear to be blocking a bit, if you looked in tokio-console, for instance.

If that's not the case, I'm happy to merge this, with the extra tracing-subscriber stuff pulled back out of the example.

Reply:
There is a single TRACE event that marks the blocking task:
```
   0.011094292s DEBUG request{uri=http://example.com/}:resolve{host=example.com}: hyper_util::client::legacy::connect::dns: resolving host="example.com"
   0.015239834s DEBUG request{uri=http://example.com/}:resolve{host=example.com}: hyper_util::client::legacy::connect::dns: close time.busy=4.15ms time.idle=227µs
   0.015319375s TRACE request{uri=http://example.com/}:runtime.spawn{kind=blocking task.name= task.id=5 fn=<hyper_util::client::legacy::connect::dns::GaiResolver as tower_service::Service<hyper_util::client::legacy::connect::dns::Name>>::call::{{closure}} loc.file="src/client/legacy/connect/dns.rs" loc.line=122 loc.col=24}: tokio::task::blocking: close time.busy=4.25ms time.idle=155µs
   0.015470667s TRACE request{uri=http://example.com/}:runtime.resource{concrete_type="Sleep" kind="timer" loc.file="src/client/legacy/connect/http.rs" loc.line=603 loc.col=28}: runtime::resource::state_update: duration=301 duration.unit="ms" duration.op="override"
   0.015578875s DEBUG request{uri=http://example.com/}: hyper_util::client::legacy::connect::http: connecting to 93.184.215.14:80
...
```
which results in a blocking task showing up briefly in tokio console:
<img width="1724" alt="image" src="https://github.com/user-attachments/assets/fbf26bcf-ac4e-42a5-9a61-27ad2ce85b9c">

But the blocked time is allocated to the blocked task, which is probably correct in that it represents the reality of what's actually happening there.

Reply:
Removed the tracing_subscriber btw.